### PR TITLE
Convert to go mod

### PR DESCRIPTION
**Pull request by davepgreene**
As `dep` is on the way out and all other Kolide projects in GitHub have migrated to use modules, I thought it made sense to convert this one too.

### CODE REVIEW

**Comment by davepgreene** [comment]
This still has some work to do to fix up the circleci stuff but I didn't have time to do it yesterday.

File: Makefile [hunk 1]
```diff
@@ -103,10 +103,10 @@ endif
 build: fleet fleetctl
 
 fleet: .prefix .pre-build .pre-fleet
-	go build -i -o build/${OUTPUT} -ldflags ${KIT_VERSION} ./cmd/fleet
+	GO111MODULE=on go build -mod vendor -i -o build/${OUTPUT} -ldflags ${KIT_VERSION} ./cmd/fleet
```
Comment (directionless) [inline Makefile]: `GO111MODULE=on` would be better set on the top level of the Makefile, akin to https://github.com/kolide/launcher/blob/master/Makefile#L11

Comment (directionless) [reply Makefile]: Why vendor here?

Comment (davepgreene) [reply Makefile]: Mainly to mimic the existing behavior with `dep`. I'll remove 😄

**Comment by davepgreene** [comment]
@directionless can you guys build an official `fleet-builder` image off of `tools/ci/docker/fleet-builder/1.11/Dockerfile`? CI won't pass until the image exists.

**Comment by zwass** [comment]
@davepgreene it should now be possible to complete this by utilizing the changes in https://github.com/kolide/fleet/pull/2081. You will no longer need to update the fleet-builder container with those changes. Thank you!

**Comment by zwass** [comment]
Hi @davepgreene apologies again for the delay on this. Are you going to complete the PR? If not I will close and open a new PR to make this migration.

**Comment by dgreene-r7** [comment]
Hey @zwass, work has picked up and I haven't had the opportunity to do anything with this PR. Feel free to take anything I've done here (or not) and move forward as it's likely I won't be able to touch it for at least a month or two.

**Comment by zwass** [comment]
Thanks @davepgreene. Hope to see more contributions from you in the future!

**Comment by zwass** [comment]
Completed this migration in https://github.com/kolide/fleet/pull/2094.

**Pull request closed without merging**